### PR TITLE
feat: add flashmint toggle

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Adds a toggle for switching different quote types (swap|flashmint) - if available
* Shows 0x quote as soon as available, loads flash mint quote in the background
* Fixes localization issues of output amount
* Uses diff to use less runs when calculating a flash mint quote (isMinting: true) to fit the user's input amount

### TODOs
- [x] test selected quote is used for tx
- [x] disable toggle when there is only one quote type available
- [x] add flash mint tooltip
- [x] improve flash mint re-quote loop

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/components/swap/components/trade-details/trade-info.tsx [hunk 1]
```diff
@@ -26,14 +33,15 @@ const TradeInfoItemRow = (props: { item: TradeInfoItem }) => {
...
+        {isLoading && <StyledSkeleton width={60} />}
+        {!isLoading && isLink === true && (
           <Link isExternal href={values[1]}>
             <Text fontSize='12px' fontWeight='700' textColor={colors.icGray3}>
               {shortenAddress(values[0])}
             </Text>
           </Link>
         )}
-        {isLink === undefined && (
+        {!isLoading && isLink === undefined && (
```
Comment (0xonramp) [inline src/components/swap/components/trade-details/trade-info.tsx:44]: Can `isLink === false` be a valid case/need to be handled too?

File: src/lib/hooks/use-best-quote/index.ts [hunk 2]
```diff
@@ -89,7 +89,71 @@ export const useBestQuote = () => {
...
+        if (canFlashmintIndexToken) {
+          console.log('canFlashmintIndexToken')
+          let dexData = null
+          if (quote0x !== null) {
+            const buyAmount = isMinting
+              ? quote0x.indexTokenAmount.toString()
+              : quote0x.inputOutputTokenAmount.toString()
+            dexData = {
+              buyAmount,
+              estimatedPriceImpact: quote0x.priceImpact?.toString() ?? '5',
```
Comment (0xonramp) [inline src/lib/hooks/use-best-quote/index.ts:102]: `estimatedPriceImpact` is in dollar terms? Something else?

**Review by 0xonramp — COMMENTED** [review]
LGTM - just a few smol questions

File: src/components/swap/components/trade-details/trade-info.tsx (hunk 1 above)
Comment (janndriessen) [reply src/components/swap/components/trade-details/trade-info.tsx:44]: Good point, will check.

**Pull request merged**